packages/kbn-docs-utils/src/api_docs/build_api_declarations/extract_import_refs.ts

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { KibanaPlatformPlugin, ToolingLog } from '@kbn/dev-utils';
+import { getApiSectionId, getPluginApiDocId, getPluginForPath } from '../utils';
+import { ApiScope, TextWithLinks } from '../types';
+
+/**
+ *
+ * @param text A string that may include an API item that was imported from another file. For example:
+ * "export type foo = string | import("kibana/src/plugins/a_plugin/public/path").Bar".
+ * @param plugins The list of registered Kibana plugins. Used to get the plugin id, which is then used to create
+ * the DocLink to that plugin's page, based off the relative path of any imports.
+ * @param log Logging utility for debuging
+ *
+ * @returns An array structure that can be used to create react DocLinks. For example, the above text would return
+ * something like:
+ * [ "export type foo = string | ", // Just a string for the pretext
+ *   { id: "a_plugin", section: "public.Bar", text: "Bar" } // An object with info to create the DocLink.
+ * ]
+ */
+export function extractImportReferences(
+  text: string,
+  plugins: KibanaPlatformPlugin[],
+  log: ToolingLog
+): TextWithLinks {
+  const texts: TextWithLinks = [];
+  let pos = 0;
+  let textSegment: string | undefined = text;
+  const max = 5;
+  while (textSegment) {
+    pos++;
+    if (pos > max) break;
+
+    const ref = extractImportRef(textSegment);
+    if (ref) {
+      const index = textSegment.indexOf('import("');
```

Comment:
I don't think so because the order is important. I need to know where to inject the link.  So the output for

`class Foo extends import("src/bar").Bar<string>`
is:
["class Foo extends ", infoToBuildBarLink, "<string>"]

So I can concatenate it all together: `pretext + link + post text`

If I used str replace I would get "class Foo extends <string>".  Unless I'm not following your suggestion? I wouldn't be surprised if I'm overcomplicating this logic.

Reply:
Gotcha. I overlooked that we were creating an array of strings. I created https://github.com/stacey-gammon/kibana/pull/17 for a few nitpicks that were hard to express using GitHub comments.